src/Features/CSharp/Portable/Braces/CSharpRemoveBracesCodeFixProvider.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Roslyn.Utilities;
+using Microsoft.CodeAnalysis.Formatting;
+
+namespace Microsoft.CodeAnalysis.CSharp.Diagnostics.AddBraces
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.RemoveBraces), Shared]
+    [ExtensionOrder(After = PredefinedCodeFixProviderNames.AddAwait)]
+    internal class CSharpRemoveBracesCodeFixProvider : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(IDEDiagnosticIds.RemoveBracesDiagnosticId);
+
+        public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            context.RegisterCodeFix(
+                new MyCodeAction(
+                    FeaturesResources.Remove_braces,
+                    c => RemoveBracesAsync(context, c)),
+                context.Diagnostics);
+
+            return SpecializedTasks.EmptyTask;
+        }
+
+        protected async Task<Document> RemoveBracesAsync(CodeFixContext context, CancellationToken cancellationToken)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var diagnostic = context.Diagnostics.First();
+            var diagnosticSpan = diagnostic.Location.SourceSpan;
+            var statement = root.FindNode(diagnosticSpan);
+
+            var newRoot = root.ReplaceNode(statement, GetReplacementNode(statement).WithAdditionalAnnotations(Formatter.Annotation));
+            return context.Document.WithSyntaxRoot(newRoot);
+        }
+
+        private SyntaxNode GetReplacementNode(SyntaxNode statement)
+        {
+            switch (statement.Kind())
+            {
+                case SyntaxKind.IfStatement:
+                    var ifSyntax = (IfStatementSyntax)statement;
+                    return GetNewNode(statement, (BlockSyntax)ifSyntax.Statement).WithTrailingTrivia(ifSyntax.Statement.GetTrailingTrivia());
```

Comment:
I can really clean some of this stuff up with dynamics, but I've heard that has a performance hit.  What do you think?